platform-operator/helm-pod/main.go

```diff
@@ -331,28 +331,31 @@ func deleteCRDInstances(request *restful.Request, response *restful.Response) {
 			helmreleaseNS, helmrelease := getHelmReleaseName(status)
 			fmt.Printf("Helm release:%s, %s\n", helmreleaseNS, helmrelease)
 			if helmreleaseNS != "" && helmrelease != "" {
-				deleteHelmRelease(helmreleaseNS, helmrelease)
-				//if ok {
+				// Do in the background as 'helm delete' and deleting NS are time consuming actions
+				go func() {
```

Comment:
I see, you have combined helm delete and NS delete in a separate function and are invoking that as a co-routine; instead of making two separate co-routine calls.